internal/cli/decommission.go

```diff
@@ -0,0 +1,124 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"slices"
+	"strings"
+
+	api "github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/internal/client"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+var (
+	// todo - add api.FactoryReset once we support it
+	allowedTargets = []string{string(api.DeviceDecommissionTargetTypeUnenroll)}
+)
+
+type DecommissionOptions struct {
+	GlobalOptions
+	DecommissionTarget string
+}
+
+func DefaultDecommissionOptions() *DecommissionOptions {
+	return &DecommissionOptions{
+		GlobalOptions:      DefaultGlobalOptions(),
+		DecommissionTarget: string(api.DeviceDecommissionTargetTypeUnenroll),
+	}
+}
+
+func NewCmdDecommission() *cobra.Command {
+	o := DefaultDecommissionOptions()
+	cmd := &cobra.Command{
+		Use:   "decommission device/NAME",
+		Short: "Decommission a device.",
+		Args:  cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := o.Complete(cmd, args); err != nil {
+				return err
+			}
+			if err := o.Validate(args); err != nil {
+				return err
+			}
+			return o.Run(cmd.Context(), args)
+		},
+		SilenceUsage: true,
+	}
+	o.Bind(cmd.Flags())
+	return cmd
+}
+
+func (o *DecommissionOptions) Bind(fs *pflag.FlagSet) {
+	o.GlobalOptions.Bind(fs)
+	fs.StringVarP(&o.DecommissionTarget, "target", "t", o.DecommissionTarget, "Specify the type of decommissioning operation: unenroll, or factoryreset")
```

Comment:
should we remove factoryreset from docs until impl?

Reply:
otherwise could be bug food :)

Reply:
You're right. Just changed it.